httpcore/_async/http11.py

```diff
@@ -340,3 +353,34 @@ async def aclose(self) -> None:
             self._closed = True
             async with Trace("response_closed", logger, self._request):
                 await self._connection._response_closed()
+
+
+class HTTP11UpgradeStream(AsyncNetworkStream):
```

Comment:
- is this only for `Upgrade`?
- is this only for http1.1?
- how about adding `Async` prefix
- Perhaps could move it to `httpcore._backends` into `sync` and `auto` submodules.